azurelinuxagent/ga/update.py

```diff
@@ -487,9 +488,44 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler):
         finally:
             self.last_incarnation = incarnation
 
+    @staticmethod
+    def __get_vmagent_upgrade_status(protocol, incarnation_changed):
+        """
+        This function gets the VMAgent upgrade status as per the last GoalState.
+        Returns: None if the last GS does not ask for requested version else VMAgentUpdateStatus
+        """
+        upgrade_status = None
+
+        try:
+            agent_manifests, _ = protocol.get_vmagent_manifests()
+            # Expectation here is that there will only be one manifest per family passed down from CRP.
+            # If there are multiple, we pick the first one
```

Comment:
what would multiple statuses indicate? an error? do we need to log that?

Reply:
maybe change the comment to say that during validation we already checked there is only 1 item so we just pick up the first one?

Reply:
sure let me rephrase

Reply:
Done